tests/plenary/state/state_spec.lua

```diff
@@ -0,0 +1,144 @@
+local utils = require('orgmode.utils')
+local state = nil
+local cache_path = vim.fs.normalize(vim.fn.stdpath('cache') .. '/org-cache.json', { expand_env = false })
+
+describe('State', function()
+  before_each(function()
+    -- Ensure the cache file is removed before each run
+    state = require('orgmode.state.state')
+    vim.wait(50, function()
+      return state._ctx.saved
+    end, 10)
+    state:wipe()
+    vim.fn.delete(cache_path, 'rf')
+  end)
+
+  after_each(function()
+    state._ctx.saved = false
+    state._ctx.loaded = false
+    state = nil
+    -- vim.fn.delete(cache_path, 'rf')
+  end)
+
+  it("should create a state file if it doesn't exist", function()
+    -- Ensure the file doesn't exist
+    vim.fn.delete(cache_path, 'rf')
```

Comment:
This is already done in `before_each` so there's no need to  delete. You can leave the check below just in case.

Reply:
Yep, will remove.

Reply:
Done.